### ISSUE TITLE
Make label and annotation keys expected to be used by users part of the _public_ API in the `api` module

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#7988]
As originally discussed in #7983 => it would make sense to have the key annotations and labels we expect to be used by users in the `api` module and not in the `operator-common` module. That would make it easier to use them when automating the API. This should for example include the `strimzi.io/cluster` label, the different restart or rolling update annotations etc.

**Comment by DevPJ9** [comment]
Hello @scholzj. I would like to work on this.

**Comment by scholzj** [comment]
I think we need to triage this first and think through where exactly the label and annotation keys should be moved. But I can ping you once it is ready to be worked on.

**Comment by DevPJ9** [comment]
Ok, sure. Thanks for the information.

**Comment by scholzj** [comment]
Triaged on 9.2.2023: Makes sense to move it to the `api` module:
* It should be one class for Labels and one for Annotations
* The constant name in the class should describe what it should be applied to (e.g. Secret, StrimziPodset, etc.)
* We should try to move all _public_ labels and annotations. Not just the `strimzi.io/cluster` label. So e.g. the manual rolling update annotations, CA renewal annotations etc.

**Comment by scholzj** [comment]
@DevPJ9 We added a bit more details about how it might be implemented. So feel free to look at it if you are still interested.

**Comment by DevPJ9** [comment]
ok @scholzj. Thanks for the update. I am looking into it.

**Comment by scholzj** [comment]
I guess that part of the task is to come up with them - at least with the initial batch. Going through the docs and looking for all the procedures where the user is instructed to label or annotate something should be a good start I think.

**Comment by AnshulAngaria** [comment]
@scholzj I would like to work on this.

**Comment by scholzj** [comment]
@AnshulAngaria: As you see above, @DevPJ9 planned to work on this. I'm not sure if this is still the case or not.

**Comment by DevPJ9** [comment]
@scholzj I will start working on it after this week. There are some things which are keeping me engaged for now. If @AnshulAngaria wants he can start working on the task and if he finishes it, that's great. So yeah, @AnshulAngaria you can work on this task.

**Comment by AnshulAngaria** [comment]
@DevPJ9 @scholzj Thanks, I will start working on this.

**Comment by pratikkolkar** [comment]
@scholzj - Can I take up this issue?. If yes, below I have stated my understanding of the issue and having some doubts.

**Problem Statement:**
Move Labels class from ```operator-common``` module to ```api``` module.

**Expected Outcome:** ??
I would appreciate it if you could provide more details on how this enhancement would be beneficial. As a beginner, I may not fully grasp the potential outcomes.

To start, I'd like to acquire some background information about the "Label" class, including its purpose and significance. Could you please suggest where I can find answers to these questions?

Please correct me if I am wrong in my understanding.

**Comment by scholzj** [comment]
Sorry, I was away and then busy with the 0.37.0 release and it seems nobody else from @strimzi/maintainers replied 😞.

I would probably expect the following from this issue:
* Create new `Labels` class in the `api` module and move the following labels there. I would keep the other labels and all the methods in the `operator-common` module.
    * STRIMZI_KIND_LABEL
    * STRIMZI_CLUSTER_LABEL
    * STRIMZI_COMPONENT_TYPE_LABEL
    * STRIMZI_NAME_LABEL
* Create a new `Annotations` class in the `api` module and move there form the `Annotations` class in `operator-common` the following constants (again, keep all the other annos and methods in `operator-common`)
    * STRIMZI_IO_CONNECT_FORCE_REBUILD
    * ANNO_STRIMZI_IO_PAUSE_RECONCILIATION
    * ANNO_STRIMZI_IO_MANUAL_ROLLING_UPDATE
    * ANNO_STRIMZI_IO_REBALANCE
    * ANNO_STRIMZI_IO_REBALANCE_AUTOAPPROVAL
    * ANNO_STRIMZI_IO_RESTART_CONNECTOR
    * ANNO_STRIMZI_IO_RESTART_TASK
    * ANNO_STRIMZI_IO_RESTART_CONNECTOR_TASK
    * ANNO_STRIMZI_IO_RESTART_CONNECTOR_TASK_PATTERN_CONNECTOR
    * ANNO_STRIMZI_IO_RESTART_CONNECTOR_TASK_PATTERN_TASK
    * ANNO_STRIMZI_IO_NEXT_NODE_IDS
    * ANNO_STRIMZI_IO_REMOVE_NODE_IDS
    * ANNO_STRIMZI_IO_NODE_POOLS
    * AbstractScalableNamespacedResourceOperator.ANNO_STRIMZI_IO_DELETE_POD_AND_PVC
    * Ca.ANNO_STRIMZI_IO_FORCE_RENEW
    * Ca.ANNO_STRIMZI_IO_FORCE_REPLACE

In other words, move there the labels and annotations that users are expected to use. WDYT @ppatierno @tombentley & @strimzi/maintainers?

**Comment by pratikkolkar** [comment]
I created ```Labels``` class and ```Annotations``` class ```io.strimzi.api.kafka``` package. Added the above mentioned labels and annotations respectively. 
As both new classes only consists string constant it didn't make sense to add test cases.

<img width="292" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/61497268/f8dea1ca-5ac9-4512-89ee-e0c4a22c1499">

Please let me know if anything else needs to be added.  If not then can I go ahead with pull request ?

**Comment by scholzj** [comment]
I wonder if this should be in the `io.strimzi.api` package. Any thoughts @ppatierno @tombentley?

**Comment by ppatierno** [comment]
I think that the list filled by Jakub is a sensible one about what should be moved.

> I wonder if this should be in the io.strimzi.api package. Any thoughts @ppatierno @tombentley?

Yeah directly under `io.strimzi.api` and not `kafka` under it.

**Comment by pratikkolkar** [comment]
Moved under `io.strimzi.api` package. Creating PR.
